### PR TITLE
chore: imgix default parameters passed via custom preference and fixe…

### PR DESCRIPTION
**Pull request by syedbadshah-toptal**
1. imgix default parameters passed via custom preference
2. Meta data updated with all the imgix custom preferences.

### CODE REVIEW

File: cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js [hunk 1]
```diff
@@ -46,10 +46,10 @@ module.exports.init = function (editor) {
   // Create a configuration for a custom editor to be displayed in a modal breakout dialog (breakout editor)
   var breakoutEditorConfig = new HashMap();
   imgixApi["apiKey"] = currentSite.getCustomPreferenceValue(
-    "imgixPageDesignerAPIkey"
+    "ImgixapiKey"
```
Comment (luqven) [inline cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js:49]: imgix should never be capitalized. It's weird I know 😆 
```suggestion
    "imgixApiKey"
```

File: cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js [hunk 2]
```diff
@@ -46,10 +46,10 @@ module.exports.init = function (editor) {
...
   );
   imgixApi["source"] = currentSite.getCustomPreferenceValue(
-    "imgixPageDesignerSourceName"
+    "ImgixSourceId"
```
Comment (luqven) [inline cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js:52]: Not sure why we're changing this to `id` from `name`. Could you provide some background on this? 

Also same as above, `imgix` should never be capitalized.

File: cartridges/int_imgix_pd/cartridge/experience/components/imgix/imageComponent.js [hunk 3]
```diff
@@ -16,7 +17,7 @@ module.exports.render = function (context, modelIn) {
   var content = context.content;
 
   // TODO: to be passed down from site settings
-  const defaultParams = "auto=format,compress&fit=crop";
+  const defaultParams = Site.getCurrent().getCustomPreferenceValue("ImgixDefaultParams");
```
Comment (luqven) [inline cartridges/int_imgix_pd/cartridge/experience/components/imgix/imageComponent.js:20]: nice 👏🏼

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 4]
```diff
@@ -1,29 +1,92 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<metadata xmlns="http://www.demandware.com/xml/impex/metadata/2006-10-31">
+<metadata xmlns="http://www.demandware.com/xml/impex/metadata/2006-10-31">
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:2]: Please remove the extra spacing added to this line.
```suggestion
<metadata xmlns="http://www.demandware.com/xml/impex/metadata/2006-10-31">
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 5]
```diff
@@ -1,29 +1,92 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<metadata xmlns="http://www.demandware.com/xml/impex/metadata/2006-10-31">
+<metadata xmlns="http://www.demandware.com/xml/impex/metadata/2006-10-31">  
     <type-extension type-id="SitePreferences">
         <custom-attribute-definitions>
-            <attribute-definition attribute-id="imgixPageDesignerAPIkey">
-                <display-name xml:lang="x-default">imgix API Key</display-name>
-                <description xml:lang="x-default">The API Key of your imgix account</description>
+			<attribute-definition attribute-id="ImgixBody">
+                <display-name xml:lang="x-default">IMGIX API BODY</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:6]: Why have all of these `display-name` (s) been changed to be all-caps? Can you provide some background for this?

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 6]
```diff
@@ -1,29 +1,92 @@
...
                 <type>string</type>
                 <mandatory-flag>false</mandatory-flag>
                 <externally-managed-flag>false</externally-managed-flag>
                 <min-length>0</min-length>
             </attribute-definition>
-            <attribute-definition attribute-id="imgixPageDesignerSourceName">
-                <display-name xml:lang="x-default">imgix Origin Name</display-name>
-                <description xml:lang="x-default">The Origin Name of your imgix account</description>
+			<attribute-definition attribute-id="ImgixHeaders">
+                <display-name xml:lang="x-default">IMGIX API HEADERS</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:13]: Why have the `description` fields been removed? I think the user could benefit from understanding what these attributes do.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 7]
```diff
@@ -1,29 +1,92 @@
...
+                <min-length>0</min-length>
+            </attribute-definition>
+			<attribute-definition attribute-id="ImgixapiKey">
+                <display-name xml:lang="x-default">IMGIX API KEY</display-name>
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+                <min-length>0</min-length>
+            </attribute-definition>
+			<attribute-definition attribute-id="ImgixMethod">
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:40]: Maybe I'm forgetting something we've discussed, but what is `ImgixMethod`?

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 8]
```diff
@@ -1,29 +1,92 @@
...
+            </attribute-definition>
+			<attribute-definition attribute-id="Imgixversion">
+                <display-name xml:lang="x-default">IMGIX VERSION</display-name>
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+                <min-length>0</min-length>
+            </attribute-definition>
+			<attribute-definition attribute-id="ImgixDefaultParams">
+                <display-name xml:lang="x-default">IMGIX VERSION</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:69]: This `display-name` seems to be incorrect. Should it not be `IMGIX DEFAULT PARAMS` is we're following the convention you've established here?

**Review by luqven — CHANGES_REQUESTED** [review]
I'm a little confused by these changes. Not sure what all of these attributes are supposed to be referencing/enabling. Moreover, I think we should include description fields on all of these attributes to ensure users know what they do.

File: cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js (hunk 1 above)
Comment (frederickfogerty) [reply cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js:49]: To clarify this suggestion, the name of the custom preference value should also be changed in the xml file

File: cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js (hunk 2 above)
Comment (frederickfogerty) [reply cartridges/bm_imgix_pd/cartridge/experience/editors/imgix/imgixEditorTrigger.js:52]: Could we probably just delete this whole line/preference? We aren't using this anywhere right?

**Review by frederickfogerty — CHANGES_REQUESTED** [review]
LGTM after Luis's comments are addresses. For now, I've marked this as "Request Changes"

**Review by frederickfogerty — APPROVED** [review]
LGTM 👍

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 9]
```diff
@@ -2,28 +2,28 @@
 <metadata xmlns="http://www.demandware.com/xml/impex/metadata/2006-10-31">
     <type-extension type-id="SitePreferences">
         <custom-attribute-definitions>
-            <attribute-definition attribute-id="imgixPageDesignerAPIkey">
-                <display-name xml:lang="x-default">imgix API Key</display-name>
-                <description xml:lang="x-default">The API Key of your imgix account</description>
+			<attribute-definition attribute-id="imgixapiKey">
+                <display-name xml:lang="x-default">Imgix API Key</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:6]: non-blocking, but lets try not to capitalize imgix.
```suggestion
                <display-name xml:lang="x-default">imgix API Key</display-name>
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 10]
```diff
@@ -2,28 +2,28 @@
...
                 <mandatory-flag>false</mandatory-flag>
                 <externally-managed-flag>false</externally-managed-flag>
                 <min-length>0</min-length>
             </attribute-definition>
-            <attribute-definition attribute-id="imgixPageDesignerSourceName">
-                <display-name xml:lang="x-default">imgix Origin Name</display-name>
-                <description xml:lang="x-default">The Origin Name of your imgix account</description>
+			<attribute-definition attribute-id="imgixDefaultParams">
+                <display-name xml:lang="x-default">Imgix Default Params</display-name>
+                <description xml:lang="x-default">Imgix Default Params to be passed in URL</description>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:15]:
```suggestion
                <description xml:lang="x-default">imgix Default Params to be passed in URL</description>
```

**Review by luqven — APPROVED** [review]

**Comment by luqven** [comment]
LGTM, we can ignore the imgix capitalization for now. Just calling it to your attention for the future.

**Pull request merged**